Services/UI/OJS.Services.Ui.Business/Implementations/UsersBusinessService.cs

```diff
@@ -1,141 +1,190 @@
-    using X.PagedList;
-
-    public class UsersBusinessService : IUsersBusinessService
+    private readonly IUsersProfileDataService usersProfileData;
+    private readonly IUserProviderService userProvider;
+
+    public UsersBusinessService(
+        IUsersProfileDataService usersProfileData,
+        IUserProviderService userProvider)
     {
-        private readonly IUsersProfileDataService usersProfileData;
-        private readonly IUserProviderService userProvider;
+        this.usersProfileData = usersProfileData;
+        this.userProvider = userProvider;
+    }
 
-        public UsersBusinessService(
-            IUsersProfileDataService usersProfileData,
-            IUserProviderService userProvider)
+    public async Task<UserAuthInfoServiceModel?> GetAuthInfo()
+    {
+        var currentUser = this.userProvider.GetCurrentUser();
+
+        if (currentUser.IsNull())
         {
-            this.usersProfileData = usersProfileData;
-            this.userProvider = userProvider;
+            return null;
         }
 
-        public async Task<UserProfileServiceModel?> GetUserProfileByUsername(string? username)
-        {
-            var currentUser = this.userProvider.GetCurrentUser();
+        var profile = await this.usersProfileData
+            .GetByIdQuery(currentUser.Id!)
+            .Include(u => u.UsersInRoles)
+            .ThenInclude(ur => ur.Role)
+            .FirstOrDefaultAsync();
 
-            if (currentUser.Id == null && (username.IsNull() || username!.IsEmpty()))
-            {
-                throw new BusinessServiceException("Empty username is not valid");
-            }
+        return profile?.Map<UserAuthInfoServiceModel>();
+    }
 
-            var userWithUsernameExists = await this.usersProfileData.Exists(p => p.UserName == username);
+    /// <summary>
+    /// Retrieves either a short or full user profile based on the logged-in user's permissions.
+    /// Only Administrators or profile owners retrieve full user profile.
+    /// User id is returned as empty string if the logged in user is not Administrator or admin.
+    /// </summary>
+    /// <param name="username">The username of the profile to retrieve.</param>
+    /// <returns>
+    /// A task that returns a <see cref="UserProfileServiceModel"/> containing the user's profile,
+    /// or null if not found or accessible.
+    /// </returns>
+    /// <exception cref="BusinessServiceException">
+    /// Thrown if the username is invalid or does not exist.
+    /// </exception>
+    public async Task<UserProfileServiceModel?> GetUserShortOrFullProfileByLoggedInUserIsAdminOrProfileOwner(string? username)
+    {
+        var currentUser = this.userProvider.GetCurrentUser();
 
-            if (!userWithUsernameExists)
-            {
-                throw new BusinessServiceException("User with this username does not exist");
-            }
+        if (currentUser.Id == null && (username.IsNull() || username!.IsEmpty()))
```

Comment:
We could simply this check:
```if (currentUser.Id == null && string.IsNullOrWhiteSpace(username))```

Reply:
Since username is nullable i cannot use an extension method over the variable before checking for its nullability first